### PR TITLE
make boomerang optional with config

### PR DESCRIPTION
**Pull request by sancar**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/config/ClientConfig.java [hunk 1]
```diff
@@ -919,6 +921,29 @@ public ClientConfig setUserContext(ConcurrentMap<String, Object> userContext) {
         return userContext;
     }
 
+    /**
+     * This feature reduces number of hops and increase performance for smart clients.
+     * It is enabled by default for smart clients. For unisocket clients this feature can not be enabled.
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/config/ClientConfig.java]: we can define what happens if set for dummy client, better to clarify `feature can not be enabled` wording.

File: hazelcast/src/test/java/com/hazelcast/client/ClientBackupAckTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    private boolean isEnabled(HazelcastInstance client) {
+        HazelcastClientInstanceImpl clientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        ClientListenerService listenerService = clientInstanceImpl.getListenerService();
+        Collection<EventHandler> values = ((SmartClientListenerService) listenerService).getEventHandlers().values();
+        boolean found = false;
+        for (EventHandler value : values) {
+            if (value instanceof SmartClientInvocationService.BackupEventHandler) {
+                found = true;
```
Comment (ihsandemir) [inline hazelcast/src/test/java/com/hazelcast/client/ClientBackupAckTest.java]: why not return here?

File: hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastClientBeanDefinitionParser.java [hunk 3]
```diff
@@ -183,6 +183,8 @@ AbstractBeanDefinition createConfigBean(Node rootNode) {
                     configBuilder.addPropertyValue("instanceName", getTextContent(node));
                 } else if ("labels".equals(nodeName)) {
                     handleLabels(node);
+                } else if ("backup-ack-to-client-enabled".equals(nodeName)) {
```
Comment (ihsandemir) [inline hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastClientBeanDefinitionParser.java]: Renaming suggestion: `backup-ack-enabled`

Comment (sancar) [reply hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastClientBeanDefinitionParser.java]: It is decided this way in design. In my opinion, backup ack enabled might be misunderstood. There is always backup ack, but without this, it is to the member, not to the client.

**Pull request merged**